integration_test/tests/helper/Expect.ts

```diff
@@ -0,0 +1,307 @@
+    },
+
+    toBeTruthy() {
+      assert(actual, `Expected ${formatValue(actual)} to be truthy`);
+    },
+
+    toBeFalsy() {
+      assert(!actual, `Expected ${formatValue(actual)} to be falsy`);
+    },
+
+    toBeDefined() {
+      assert(
+        actual !== undefined,
+        `Expected ${formatValue(actual)} to be defined`
+      );
+    },
+
+    toBeUndefined() {
+      assert(
+        actual === undefined,
+        `Expected ${formatValue(actual)} to be undefined`
+      );
+    },
+
+    toBeNull() {
+      assert(actual === null, `Expected ${formatValue(actual)} to be null`);
+    },
+
+    toNotBeNull() {
+      assert(actual !== null, `Expected ${formatValue(actual)} to not be null`);
+    },
+
+    toBeInstanceOf(expected: any) {
+      assert(
+        actual instanceof expected,
+        `Expected ${formatValue(actual)} to be instance of ${
+          expected.name || 'unknown'
+        }`
+      );
+    },
+
+    toContain(expected: any) {
+      assert(
+        actual && actual.includes && actual.includes(expected),
+        `Expected ${formatValue(actual)} to contain ${formatValue(expected)}`
+      );
+    },
+
+    toHaveLength(expected: number) {
+      const actualLength = actual ? actual.length : undefined;
+      assert(
+        actual && actual.length === expected,
+        `Expected ${formatValue(
+          actual
+        )} to have length ${expected}, but got ${actualLength}`
+      );
+    },
+
+    toHaveProperty(property: string) {
+      assert(
+        actual && Object.prototype.hasOwnProperty.call(actual, property),
+        `Expected ${formatValue(actual)} to have property '${property}'`
+      );
+    },
+
+    toHavePropertyValue(property: string, value: any) {
+      const actualValue = actual ? actual[property] : undefined;
+      assert(
+        actual &&
+          Object.prototype.hasOwnProperty.call(actual, property) &&
+          actual[property] === value,
+        `Expected ${formatValue(
+          actual
+        )} to have property '${property}' with value ${formatValue(
+          value
+        )}, but got ${formatValue(actualValue)}`
+      );
+    },
+
+    toMatchRegex(regex: RegExp) {
+      assert(
+        regex.test(String(actual)),
+        `Expected ${formatValue(actual)} to match regex ${regex}`
+      );
+    },
+
+    toBeCloseTo(expected: number, precision: number = 2) {
+      if (!isNumber(actual) || !isNumber(expected)) {
+        throwErr(
+          `Both values must be numbers: actual=${formatValue(
+            actual
+          )}, expected=${formatValue(expected)}`
+        );
+      }
+      const diff = Math.abs(actual - expected);
+      const tolerance = Math.pow(10, -precision);
+      assert(
+        diff <= tolerance,
+        `Expected ${formatValue(actual)} to be close to ${formatValue(
+          expected
+        )} (precision: ${precision}), but difference is ${diff}`
+      );
+    },
+
+    // Async assertion support
+    async toResolve() {
```

Comment:
Non-promise values seem to pass this check as well.

I would propose to type-check this function too to avoid silently accepting non-promises.

```suggestion
    async toResolve() {
      if (!(actual instanceof Promise)) {
        throwErr(`Expected a Promise to resolve, but got a non-promise: ${actual}`);
      }
```

---

Without type-checking:

```ts
async function testToResolve(actual: any) {
  try {
    const result = await Promise.resolve(actual);
    console.error(`Expected resolve, got: ${result}`);
  } catch (error) {
    console.log(`Expected promise to resolve, but it rejected with: ${error}`);
  }
}

(async () => {
  console.log("\nCase 1: Rejected promise");
  await testToResolve(Promise.reject("boom")); // Rejected

  console.log("\nCase 2: Resolved promise");
  await testToResolve(Promise.resolve("ok")); // Resolved

  console.log("\nCase 3: Non-promise value");
  await testToResolve(123); // Resolved   <----
})();
```

Reply:
fixed in 8faf3e73